operations/commit_queue.go

```diff
@@ -117,3 +178,124 @@ func deleteCommitQueueItem(ctx context.Context, client client.Communicator, proj
 
 	return nil
 }
+
+type mergeParams struct {
+	projectID   string
+	ref         string
+	id          string
+	pause       bool
+	description string
+	skipConfirm bool
+	large       bool
+}
+
+func (p *mergeParams) mergeBranch(ctx context.Context, confPath string) error {
+	conf, err := NewClientSettings(confPath)
+	if err != nil {
+		return errors.Wrap(err, "problem loading configuration")
+	}
+
+	ac, _, err := conf.getLegacyClients()
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	client := conf.GetRestCommunicator(ctx)
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	if p.id == "" {
+		if p.projectID == "" {
+			p.projectID = conf.FindDefaultProject()
+		} else {
+			if conf.FindDefaultProject() == "" &&
+				!p.skipConfirm && confirm(fmt.Sprintf("Make %v your default project?", p.projectID), true) {
```

Comment:
do we need to duplicate this default setting behavior here?

Reply:
Refactored validating the project id in patches and reused it.